src/benchmarking/tool_metrics/calculator.py

```diff
@@ -0,0 +1,51 @@
+from pathlib import Path
+from typing import Any
+
+from src.benchmarking.base_logger import BaseLogger
+from src.benchmarking.baseline import DialogueBaseline
+from src.benchmarking.tool_metrics.base_evaluator import BaseEvaluator
+from src.summarize_algorithms.core.base_dialogue_system import BaseDialogueSystem
+from src.summarize_algorithms.core.models import DialogueState, MetricState, Session
+
+
+class Calculator:
+    def __init__(self, logger: BaseLogger, path_to_save: str | Path | None = None) -> None:
+        self.path_to_save: str | Path | None = path_to_save
+        self.logger = logger
+
+    def evaluate(
+            self,
+            algorithms: list[BaseDialogueSystem | DialogueBaseline],
```

Comment:
make `algorithms` take not `BaseDialogueSystem | DialogueBaseline` but a protocol that has only one function, `process_dialogue`